TestControllers/TestController.psm1

```diff
@@ -569,13 +570,22 @@ Class TestController
 					$executionCount += 1
 					if (!$vmData -or $tcDeployVM) {
 						# Deploy the VM for the setup
-						$vmData = $this.TestProvider.DeployVMs($this.GlobalConfig, $this.SetupTypeTable[$setupType], $this.SetupTypeToTestCases[$key][0], `
+						$deployVMStatus = $this.TestProvider.DeployVMs($this.GlobalConfig, $this.SetupTypeTable[$setupType], $this.SetupTypeToTestCases[$key][0], `
 							$this.TestLocation, $this.RGIdentifier, $this.UseExistingRG, $this.ResourceCleanup)
+						$vmData = $null
+						$deployErrors = ""
+						if ($deployVMStatus) {
+							$vmData = $deployVMStatus
+							$deployErrors = $deployVMStatus.Error
+							if ($deployVMStatus.Keys -and ($deployVMStatus.Keys -contains "VmData")) {
+								$vmData = $deployVMStatus.VmData
+							}
```

Comment:
Just double confirm, if a TestProvider.DeployVMs() is not changed and returns the $vmData object only, it's still compatible and well handled

Reply:
yes, that s why I did this checks, to make sure the code is backward compatible with providers which do not implement

Reply:
We should add functionality to trickle down the specific reason for failing to deploy VMs for instance on azure we could grab the failed deployment steps and return the messages or check azure kernel panic

Reply:
will add the new functionality requests as issues, thanks.

Reply:
@stefanlupsa @juhlee-microsoft  Added the new issue to be solved by a different PR after this gets merged https://github.com/LIS/LISAv2/issues/308